tiledb/sm/c_api/tiledb.cc

```diff
@@ -2752,7 +2752,9 @@ int32_t tiledb_query_condition_combine(
   // Sanity check
   if (sanity_check(ctx) == TILEDB_ERR ||
       sanity_check(ctx, left_cond) == TILEDB_ERR ||
-      sanity_check(ctx, right_cond) == TILEDB_ERR)
+      (combination_op != TILEDB_NOT &&
+       sanity_check(ctx, right_cond) == TILEDB_ERR) ||
+      (combination_op == TILEDB_NOT && right_cond != nullptr))
```

Comment:
I am not sure I like using one half of a binary operator to be a unary operator.  OTOH, not sure if it is worth changing if all we are going to have is AND, OR, NOT.

Reply:
I had this same exact thought. My first reaction was to do away with the existing implementation and then provide a `tiledb_query_condition_negate` function. I polled the language binding crowd and the only responses I got were in favor of keeping things as is, reusing `tiledb_query_condition_combine`.

If I had it to do all over again, I would change the combine API from:

```C++
TILEDB_EXPORT int32_t tiledb_query_condition_combine(
    tiledb_ctx_t* ctx,
    const tiledb_query_condition_t* left_cond,
    const tiledb_query_condition_t* right_cond,
    tiledb_query_condition_combination_op_t combination_op,
    tiledb_query_condition_t** combined_cond) TILEDB_NOEXCEPT;
```

to:

```C++
TILEDB_EXPORT int32_t tiledb_query_condition_combine(
    tiledb_ctx_t* ctx,
    const tiledb_query_condition_t** conditions,
    size_t num_conditions,
    tiledb_query_condition_combination_op_t combination_op,
    tiledb_query_condition_t** combined_cond) TILEDB_NOEXCEPT;
```

The second definition removes the baked in 2-arity semantics of the `combination_op` so that `NOT` and `>2`-arity AND/OR operators would also be naturally expressed. I briefly considered combining this approach with #3814 but @eric-hughes-tiledb had some other concerns around the general approach of #3814 so I'm leaving it be for now.